lib/utils/utils.go

```diff
@@ -32,6 +32,36 @@ import (
 	"github.com/pborman/uuid"
 )
 
+// ParseAdvertiseAddress validates advertise address,
+// makes sure it's not an unreachable or multicast address
+// returns address split into host and port, port could be empty
+// if not specified
+func ParseAdvertiseAddr(advertiseIP string) (string, string, error) {
+	advertiseIP = strings.TrimSpace(advertiseIP)
+	host := advertiseIP
+	port := ""
+	if strings.Contains(advertiseIP, ":") {
```

Comment:
This will match an IPv6 address, and the will fail on net.SplitHostPort below if the address does not contain port. Is that expected behavior?

Reply:
good point, can you take another look?